beacon-chain/light-client-network.md

```diff
@@ -0,0 +1,200 @@
+# Beacon Chain Light Client Network
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+This document is the specification for the Portal Network overlay network that supports the on-demand availability of Beacon Chain light client data.
+
+## Overview
+
+A beacon chain light client could keep track of the chain of beacon block headers by performing Light client state updates
+following the light client [sync protocol](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md).
+The [LightClientBootstrap](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#lightclientbootstrap) structure allow setting up a
+[LightClientStore](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#lightclientstore) with the initial sync committee and block header from a user-configured trusted block root.
+
+Once the client establishes a recent header, it could sync to other headers by processing objects of type [LightClientUpdate](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#lightclientupdate),
+[LightClientFinalityUpdate](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#lightclientfinalityupdate)
+and [LightClientOptimisticUpdate](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#lightclientoptimisticupdate).
+These data types allow a client to stay up-to-date with the beacon chain.
+
+The Beacon Chain Light Client network is a [Kademlia](https://pdos.csail.mit.edu/~petar/papers/maymounkov-kademlia-lncs.pdf) DHT that forms an overlay network on top of
+the [Discovery v5](https://github.com/ethereum/devp2p/blob/master/discv5/discv5-wire.md) network. The term *overlay network* means that the light client network operates
+with its routing table independent of the base Discovery v5 routing table and uses the extensible `TALKREQ` and `TALKRESP` messages from the base Discovery v5 protocol for communication.
+
+The `TALKREQ` and `TALKRESP` protocol messages are application-level messages whose contents are specific to the Beacon Chain Light Client network. We specify these messages below.
+
+The Beacon Chain Light Client network uses a modified version of the routing table structure from the Discovery v5 network and the lookup algorithm from section 2.3 of the Kademlia paper.
+
+### Gossip Algorithm
```

Comment:
when you talk about gossip, you mean our portal home grown gossip (offer/accept) or rather libp2p gossip which is employed in beacon [chain](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#the-gossip-domain-gossipsub)  ?

Maybe more general question here is should any participant in portal light client netowrk be also paritcipant in gossipbupsub network ?

Reply:
> when you talk about gossip, you mean our portal home grown gossip (offer/accept) or rather libp2p gossip which is employed in beacon [chain](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#the-gossip-domain-gossipsub) ?

Here we mean only the portal offer/accept gossip mechanism.

> Maybe more general question here is should any participant in portal light client netowrk be also paritcipant in gossipbupsub network ?

I think the goal here is to eliminate the need for a light client to participate in libp2p network and use the portal network as the only networking layer.